onnx/test/parser_test.py

```diff
@@ -119,6 +119,205 @@ def test_parse_function_with_attributes(self) -> None:
+                                start_h = Add (minus_one, step_h_half)
+                                grid_h_then = Range (start_h, one, step_h)
+
+                                step_w_half = Div (step_w, two)
+                                start_w = Add (minus_one, step_w_half)
+                                grid_w_then = Range (start_w, one, step_w)
+                            },
+                            else_branch = h2 () => (float[D] grid_d_else, float[H] grid_h_else, float[W] grid_w_else) {
+                                D_float_nimus_one = Sub (D_float, one)
+                                H_float_nimus_one = Sub (H_float, one)
+                                W_float_nimus_one = Sub (W_float, one)
+                                step_d = Div (two, D_float_nimus_one)
+                                step_h = Div (two, H_float_nimus_one)
+                                step_w = Div (two, W_float_nimus_one)
+                                epsilon = Constant <value = float {1e-6}>()
+                                one_plus_epsilon = Add (one, epsilon)
+                                grid_d_else = Range (minus_one, one_plus_epsilon, step_d)
+                                grid_h_else = Range (minus_one, one_plus_epsilon, step_h)
+                                grid_w_else = Range (minus_one, one_plus_epsilon, step_w)
+                            }
+                        >
+                        ones_H_W_D = Transpose <perm = [1, 2, 0]> (ones_D_H_W)
+                        grid_d_1 = Mul(ones_H_W_D, grid_d_0)
+                        grid_d = Transpose <perm = [2, 0, 1]>(grid_d_1)
+
+                        ones_D_W_H = Transpose <perm = [0, 2, 1]> (ones_D_H_W)
+                        grid_h_1 = Mul(ones_D_W_H, grid_h_0)
+                        grid_h = Transpose <perm = [0, 2, 1]>(grid_h_1)
+
+                        grid_w = Add (grid_w_0, zero_D_H_W)
+
+                        original_grid_seq = SequenceConstruct (grid_w, grid_h, grid_d, ones_D_H_W)
+                        original_grid = ConcatFromSequence <axis: int=-1, new_axis: int=1> (original_grid_seq)
+                        constant_shape_DHW_4 = Constant <value_ints: ints = [-1, 4]> ()
+                        original_grid_DHW_4 = Reshape (original_grid, constant_shape_DHW_4)
+                        original_grid_4_DHW = Transpose (original_grid_DHW_4)
+
+                        grid_N_3_DHW = MatMul (theta, original_grid_4_DHW)
+                        grid_N_DHW_3 = Transpose <perm = [0, 2, 1]> (grid_N_3_DHW)
+                        N_D_H_W_3_seq = SequenceConstruct (N, D, H, W, int_three_1d)
+                        N_D_H_W_3 = ConcatFromSequence <axis: int=-1, new_axis: int=0> (N_D_H_W_3_seq)
+                        grid_3d_else = Reshape(grid_N_DHW_3, N_D_H_W_3)
+                        }
+                    >
+           }
+        """
+
+        model = onnx.parser.parse_model(input)
```

Comment:
Instead of duplicating the function-definition, would it be better to get the function-definition from the op-schema, and test it?

Reply:
Good point. Only one test body for AffineGrid shall exist which is in the op-schema.